src/main/java/network/brightspots/rcv/CommonDataFormatReader.java

```diff
@@ -132,6 +159,25 @@ void parseXml(List<CastVoteRecord> castVoteRecords)
       CastVoteRecordReport cvrReport = xmlMapper.readValue(inputStream, CastVoteRecordReport.class);
       inputStream.close();
 
+      // Some checks to provide nicer error messages.
+      // This is common with Unisyn's CDF CVR, which is not compatible with RCTab.
+      if (cvrReport.GpUnit == null) {
+        Logger.severe("Field \"GPUnit\" missing from CDF CVR file! "
+                + "This is common with older, unsupported formats.");
+        throw new CvrParseException();
+      }
+
+      // These fields are also required, but we are not aware of any standard format where
+      // they would be missing, so we group all these checks together.
```

Comment:
From here down, it seems to me like it might be slightly more helpful to throw a specific error for each field, even if we don't know a standard format where they might be missing.

The way I'd probably design it is to initialize a `fieldsAreMissing = false` variable, then go through each check and set it to true if any trigger. Then at the very end I'd throw the `CvrParseException` if it's `true`. That way we can clearly see every field that's missing laid out in the logs.

If you feel strongly otherwise though, feel free to leave as-is.